proposed/extended-coding-style-guide.md

```diff
@@ -163,6 +163,19 @@ class FooBar
 
 ```
 
+Compound namespaces with a depth of two or more MUST not be used. Therefore the
```

Comment:
I disagree, I had to ask Michael if this was an anti-example. I think being clear that `SubnamespaceOne\ClassB` is an example of a "compound namespace with a depth of one" is important.